Avoid NullPointerException with empty sub-communities.

hcayless opened the pull request:
If there are sub-communities that contain nothing, a community metadata export will fail with a NullPointerException, because the buildFromCommunity method will return null when called recursively and then the code will try to add that null to the result Iterator. This patch just checks whether the buildFromCommunity call returned null before trying to append it, avoiding the exception.

hcayless committed:
Avoid NullPointerException with empty communities.

mwoodiupui commented:
That should fix the immediate problem.  A more rugged fix would be to instead have addItemsToResult immediately return 'result' if 'items' is null.  That way, if it is ever called elsewhere, we don't have to remember to check for null before calling.

(buildFromCommunity *should* have returned an empty Iterator rather than null....)

hcayless commented:
That would work fine. I can modify the pull request to do that instead if you'd prefer.

mwoodiupui commented:
Yes, please do.

hcayless committed:
Refactored based on feedback.

hcayless committed:
Fixed extraneous whitespace.

hcayless commented:
addItemsToResult looked sort of unbalanced when I added the new condition and return, so I rewrote it without the if/else. If items is null, it immediately returns result; if result is empty it immediately returns items, and if it makes it past those conditions, it returns items added to result.
